tests/test_app/test_typing.py

```diff
@@ -86,6 +86,13 @@ def test_hints_from_strings():
     assert got == [SerialisableType, AlignedSeqsType]
 
 
+def test_hints_resolved_from_str():
+    got = get_constraint_names("DnaSequence")
+    assert got == {"DnaSequence"}
+    got = get_constraint_names(Union[SerialisableType, "DnaSequence"])
+    assert got == {"SerialisableType", "DnaSequence"}
+
```

Comment:
Can we add all types here as parametrized test to see if all of them correctly resolved?
I can add this after you merge this PR.